docs/advanced/newsletter_templates.md

```diff
@@ -0,0 +1,64 @@
+# Newsletter templates
+
+The newsletter templates allow the user to select a template amongst a set of of them, and use it as base to send newsletters. This allow for more customization on newsletter, tematic newsletters, etc.
+
+Code-wise, they use the content blocks system internally, so [check the docs](https://github.com/decidim/decidim/blob/master/docs/advanced/content_blocks.md) for that section first.
+
+# Adding a new template
+
+You'll first need to register the template as a content block, but specifying `:newsletter_template` as its scope:
+
+```ruby
+Decidim.content_blocks.register(:newsletter_template, :my_template) do |content_block|
+  content_block.cell "decidim/newsletter_templates/my_template"
+  content_block.settings_form_cell = "decidim/newsletter_templates/my_template_settings_form"
+  content_block.public_name_key "decidim.newsletter_templates.my_template.name"
+
+  content_block.images = [
+    {
+      name: :main_image,
+      uploader: "Decidim::NewsletterTemplateImageUploader",
+      preview: -> { ActionController::Base.helpers.asset_path("decidim/placeholder.jpg") }
+    }
+  ]
+
+  content_block.settings do |settings|
+    settings.attribute(
+      :body,
+      type: :text,
+      translated: true,
+      preview: -> { ([I18n.t("decidim.newsletter_templates.my_template.body_preview")] * 100).join(" ") }
+    )
+  end
+end
+```
+
+You'll need to add this into an initializer. Note that if you're adding this from a module, then you need to add it from the `engine.rb` file (check the docs for content blocks for more info).
+
+This is the simplest template. It has a single attribute, in this case a translatable chunk of text. Let's go line by line.
+
+Inside the block, first we define the path of the cell we'll use to render the email publicly. This cell will receive the `Decidim::ContentBlock` object, which will contain the attributes and any image we have (one in this example). In order to render cells, please note that emails have a very picky HTML syntax, so we suggest using some specialized tools to design the template, export the layout to HTML and render that through the cell. We suggest you make this cell inherit from `Decidim::NewsletterTemplates::BaseCell` for convenience.
+
+Then we define the cell that will be used to render the form in the admin section. This form needs to show inputs for the attributes defined when registering the template. It will receive the `form` object to render the input fields. We suggest this cell to inherit from `Decidim::NewsletterTemplates::BaseSettingsFormCell`.
+
+In the third line inside the block we define the I18n path to the public name of the template. This name will serve as identifier for the users, so be sure to make it descriptive.
```

Comment:
when you say the public name of the template, you mean only in the admin no?
Maybe we can rename it from `public_name_key` to `template_name_i18n_key`?

Reply:
This would imply changing the API for content blocks, which could break external integrations... Renaming this key to `template_name_i18n_key` would link the the key with the concept of templates (newsletter templates), but then wouldn't work for homepage content blocks... Homepage content blocks also have this key, but they're only seen in the admin too.

Here, "public" means "seen by users" (differentiating users from devs).